include/boost/url/impl/url.ipp

```diff
@@ -1515,49 +1519,25 @@ param(
     std::size_t i) const noexcept ->
         raw_param
 {
-    auto const make_param =
-    [this](
-        std::size_t pos,
-        std::size_t n)
-    {
-        string_view s(s_ + pos, n);
-        auto i = s.find_first_of('=');
-        if(i == string_view::npos)
-            return raw_param{ pos, n, 0 };
-        return raw_param{ pos, i, n - i };
-    };
-
     if(nparam_ == 0)
         return { offset(id_query), 0, 0 };
     if(i == nparam_)
         return { offset(id_frag), 0, 0 };
     BOOST_ASSERT(i <= nparam_);
-    auto n = len(id_query);
-    if(nparam_ < 2)
-        return make_param(
-            offset(id_query), n);
-    auto it = s_ + offset(id_query);
-    auto start = it;
-    auto const last =
-        s_ + offset(id_frag);
-    BOOST_ASSERT(n > 0);
-    for(;;)
-    {
-        for(;;)
-        {
-            ++it;
-            if(it == last)
-                break;
-            if(*it == '&')
-                break;
-        }
-        if(i == 0)
-            break;
-        start = it;
-        --i;
-    }
-    return make_param(
-        start - s_, it - start);
+
+    detail::const_params_table t(param_tab_begin());
+    string_view q = encoded_query();
+    auto const& e = t[i];
+    const char* start = q.data() + e.kp - 1;
+    const char* end = s_ + offset(id_frag);
+    if (i != nparam_ - 1)
+        end = q.data() + t[i + 1].kp - 1;
+
+    std::size_t param_pos = start - s_;
+    std::size_t param_n = end - start;
```

Comment:
I would use auto

Reply:
`auto param_pos = start - s_;` would become `long param_pos = = start - s_;` and fail in the `raw_param` constructor.

```
Non-constant-expression cannot be narrowed from type 'long' to 'std::size_t' (aka 'unsigned long') in initializer list
```